src/msix/msix.cpp

```diff
@@ -380,7 +368,24 @@ MSIX_API HRESULT STDMETHODCALLTYPE PackBundle(
     ThrowHrIfFailed(factory->CreateBundleWriter(stream.Get(), bundleVersion, &bundleWriter));
     bundleWriter4 = bundleWriter.As<IAppxBundleWriter4>();
 
-    bundleWriter4.As<IBundleWriter>()->ProcessBundlePayload(from, flatBundle);
+    if(directoryPath != nullptr && outputBundle != nullptr)
+    {
+        auto from = MSIX::ComPtr<IDirectoryObject>::Make<MSIX::DirectoryObject>(directoryPath);
+        bundleWriter4.As<IBundleWriter>()->ProcessBundlePayload(from, flatBundle);
+    }
+    else if(mappingFile != nullptr && outputBundle != nullptr)
+    {
+        MSIX::MappingFileParser mappingFileParser;
+        mappingFileParser.ParseMappingFile(mappingFile);
+        if(!mappingFileParser.IsSectionFound(MSIX::SectionID::FilesSection))
+        {
+            std::ostringstream errorBuilder;
+            errorBuilder << "The mapping file " << mappingFile << " is missing a [Files] section.";
+            ThrowErrorAndLog(MSIX::Error::BadFormat, errorBuilder.str().c_str());
+        }
+        bundleWriter4.As<IBundleWriter>()->ProcessBundlePayloadFromMappingFile(mappingFileParser.GetFileList(), mappingFileParser.GetExternalPackagesList(), flatBundle);
+    }

```

Comment:
>} [](start = 4, length = 1)

i think we should fail with invalid args in the else

Reply:
I modified the code different from how I had it before to handle other cases as well.